semgrep-core/src/matching/Matching_generic.ml

```diff
@@ -251,6 +251,7 @@ let rec equal_ast_binded_code (config : Config_semgrep.t) (a : MV.mvalue)
         MV.E { e = B.L b_lit; _ } )
       when config.constant_propagation ->
         G.equal_literal a_lit b_lit
+    | MV.Id ((s1, _), None), MV.Id ((s2, _), Some _) -> s1 = s2
```

Comment:
This may deserve a comment to explain why we need this case. It could also include a reference to the ticket.